CONTRIBUTING.md

```diff
@@ -1,12 +1,379 @@
+
+- Signature -- To identify and apply the correct quirk
+- Replacement -- To allow Zigpy and ZHA to correctly work with the device
+- device_automation_triggers - To let the Home Assistant Device Automation engine and users interact with the device
+
+### Signature
+
+The signature on a quirk identifies the device as the manufacturer implemented it. You can think of it as a fingerprint or the dna of the device. The signature is what we use to identify the device. If any part of the signature doesn't match what the device returns during discovery the quirk will not match and as a result it will not be applied. The signature is made up of several parts:
+
+- `models_info`
+- `endpoints`
+
+Models info tells the application which devices should use this particular quirk. Endpoints are the simple descriptors that we spoke about earlier exactly as they are on the device. `endpoints` is a dict where the key is the id of the endpoint and the value is an object with the following properties: `profile_id`, `device_type`, `input_clusters` and `output_clusters`. Creating the signature element is generally just a job of transcribing what the device gives us. Here is an example:
+
+```python
+signature = {
+    MODELS_INFO: [(LUMI, "lumi.plug.maus01")],
+    ENDPOINTS: {
+        # <SimpleDescriptor endpoint=1 profile=260 device_type=81
+        # device_version=1
+        # input_clusters=[0, 4, 3, 6, 16, 5, 10, 1, 2, 2820]
+        # output_clusters=[25, 10]>
+        1: {
+            PROFILE_ID: zha.PROFILE_ID,
+            DEVICE_TYPE: zha.DeviceType.SMART_PLUG,
+            INPUT_CLUSTERS: [
+                Basic.cluster_id,
+                PowerConfiguration.cluster_id,
+                DeviceTemperature.cluster_id,
+                Groups.cluster_id,
+                Identify.cluster_id,
+                OnOff.cluster_id,
+                Scenes.cluster_id,
+                BinaryOutput.cluster_id,
+                Time.cluster_id,
+                ElectricalMeasurement.cluster_id,
+            ],
+            OUTPUT_CLUSTERS: [Ota.cluster_id, Time.cluster_id],
+        },
+    },
+}
+```
+
+### Replacement
+
+The replacement on a quirk is what we want the device to be. Remember, we said that quirks were like Google translate... you can think of the replacement like the output from Google translate. The replacement dict is what will actually be used by Zigpy and ZHA to interact with the device. The structure of `replacement` is the same as signature with 2 key differences: `models_info` is generally omitted and there is an extra element `skip_configuration` that instructs the application to skip configuration if necessary. Some manufacturers have not implemented the specifications correctly and the devices come pre-configured and therefore the configuration calls fail (non Zigbee 3.0 Xiaomi devices for instance) Here is an example:
+
+```python
+replacement = {
+    SKIP_CONFIGURATION: True,
```

Comment:
`SKIP_CONFIGURATION = True` means the ZHA wont bind and configure ZCL attribute reporting for supported (by ZHA) clusters. This is a workaround for Xioami legacy devices and is not supposed to be used with any other compliant devices. Legacy xiaomi devices are hardcoded to report attributes to NWK 0x0000 address -- coordinator and  because those devices are too "sleepy" often binding/attribute report configuration was timing out for those, so this hack was introduced. 
If `SKIP_CONFIGURATION = False` or just not present on the custom cluster, then ZHA will bind and configure attribute reporting for that cluster.